src/pages/index.js

```diff
@@ -1,75 +1,183 @@
+    "Content-Type": "application/json"
+  }
+});
+
+// UserInfo instance
+const userInfo = new UserInfo({
+  nameSelector: ".profile__title",
+  descriptionSelector: ".profile__description",
+  avatarSelector: ".profile__image"
+});
+
+// Popup instances
+const popupWithImage = new PopupWithImage("#modal-preview-picture");
+popupWithImage.setEventListeners();
+
+const popupWithFormProfile = new PopupWithForm("#profile-edit-modal", handleProfileFormSubmit);
+popupWithFormProfile.setEventListeners();
+
+const popupWithFormAddCard = new PopupWithForm("#add-card-modal", handleAddCardFormSubmit);
+popupWithFormAddCard.setEventListeners();
+
+const popupWithFormAvatar = new PopupWithForm("#edit-avatar-modal", handleAvatarFormSubmit);
+popupWithFormAvatar.setEventListeners();
+
+const popupWithConfirmation = new PopupWithConfirmation({
+  popupSelector: '#delete-confirmation-modal',
+  handleConfirm: handleDeleteCard
+});
+popupWithConfirmation.setEventListeners();
+
+// Card section instance
+const cardSection = new Section(
+  {
+    renderer: (cardData) => {
+      const cardElement = createCard(cardData);
+      cardSection.addItem(cardElement);
+    }
+  },
+  ".cards__list"
+);
+
+// Load user information and initial cards
+Promise.all([api.getUserInfo(), api.getInitialCards()])
+  .then(([userData, initialCards]) => {
+    userInfo.setUserInfo({
+      name: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+    cardSection.renderItems(initialCards);
+  })
+  .catch((err) => {
+    console.error("Error loading initial data:", err);
+  });
+
+// Handle profile form submission
+function handleProfileFormSubmit(formData) {
+  popupWithFormProfile.renderLoading(true); 
+  return api.updateUserInfo(formData)
+    .then((updatedData) => {
+      userInfo.setUserInfo({
+        name: updatedData.name,
+        description: updatedData.about,
+        avatar: updatedData.avatar
+      });
+      popupWithFormProfile.close();
+      popupWithFormProfile.form.reset();
+      formValidators['profile-form'].toggleButtonState();
+    })
+    .catch((err) => {
+      console.error("Error updating user profile:", err);
+    })
+    .finally(() => {
+      popupWithFormProfile.renderLoading(false); 
+    });
+}
+
+// Handle add card form submission
+function handleAddCardFormSubmit(formData) {
+  return api.addCard({
+    name: formData.title,
+    link: formData.url
+  })
+  .then((newCard) => {
+    const cardElement = createCard(newCard);
+    cardSection.addItem(cardElement);
+    popupWithFormAddCard.form.reset();
+    popupWithFormAddCard.close();
+    formValidators['card-form'].toggleButtonState();
+  })
+  .catch((err) => {
+    console.error("Error adding card:", err);
+  })
+  .finally(() => {
+    popupWithFormAddCard.renderLoading(false);
+  });
+}
+
+// Handle avatar form submission
+function handleAvatarFormSubmit(formData) {
+  popupWithFormAvatar.renderLoading(true); 
```

Comment:
`.renderLoading(true)` is not needed in `index.js` because you have it inside `PopupWithForm`